buildSrc/src/main/java/org/elasticsearch/gradle/JdkDownloadPlugin.java

```diff
@@ -37,44 +37,46 @@
 import org.gradle.api.file.RelativePath;
 import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.Copy;
+import org.gradle.api.tasks.TaskProvider;
 
-import java.io.File;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
 import java.util.concurrent.Callable;
-import java.util.function.Supplier;
-import java.util.regex.Matcher;
+import java.util.stream.StreamSupport;
+
+import static org.elasticsearch.gradle.Util.capitalize;
+import static org.elasticsearch.gradle.tool.Boilerplate.findByName;
+import static org.elasticsearch.gradle.tool.Boilerplate.maybeCreate;
 
 public class JdkDownloadPlugin implements Plugin<Project> {
 
     private static final String REPO_NAME_PREFIX = "jdk_repo_";
-    private static final String CONTAINER_NAME = "jdks";
+    private static final String EXTENSION_NAME = "jdks";
 
     @Override
     public void apply(Project project) {
-        NamedDomainObjectContainer<Jdk> jdksContainer = project.container(Jdk.class, name -> new Jdk(name, project));
-        project.getExtensions().add(CONTAINER_NAME, jdksContainer);
+        NamedDomainObjectContainer<Jdk> jdksContainer = project.container(
+            Jdk.class,
+            name -> new Jdk(name, project.getConfigurations().create("jdk_" + name), project.getObjects())
+        );
+        project.getExtensions().add(EXTENSION_NAME, jdksContainer);
 
         project.afterEvaluate(p -> {
             for (Jdk jdk : jdksContainer) {
                 jdk.finalizeValues();
-                String vendor = jdk.getVendor();
-                String version = jdk.getVersion();
-                String platform = jdk.getPlatform();
 
                 // depend on the jdk directory "artifact" from the root project
                 DependencyHandler dependencies = project.getDependencies();
                 Map<String, Object> depConfig = new HashMap<>();
                 depConfig.put("path", ":"); // root project
-                depConfig.put("configuration", configName("extracted_jdk", vendor, version, platform));
-                dependencies.add(jdk.getConfiguration().getName(), dependencies.project(depConfig));
+                depConfig.put("configuration", configName("extracted_jdk", jdk.getVendor(), jdk.getVersion(), jdk.getPlatform()));
+                project.getDependencies().add("jdk_" + jdk.getName(), dependencies.project(depConfig));
```

Comment:
Can we either have a shared method for forming the name, or have a getConfigurationName() method on the jdk?

Reply:
I'ved pulled out a `getConfigurationName()` method as suggested.